core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/Run.java

```diff
@@ -48,250 +100,543 @@
  * @author Rob Davey
  * @since 0.0.2
  */
-@JsonSerialize(typing = JsonSerialize.Typing.STATIC)
-@JsonInclude(JsonInclude.Include.NON_NULL)
-@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "@class")
-@JsonIgnoreProperties({ "securityProfile", "submissionDocument" })
-public interface Run
-    extends SecurableByProfile, Comparable<Run>, Reportable<Run>, Watchable, Deletable, Nameable, Alertable, ChangeLoggable, Aliasable,
-    Serializable {
+@Entity
+@Table(name = "Run")
+@Inheritance(strategy = InheritanceType.JOINED)
+public class Run
+    implements SecurableByProfile, Comparable<Run>, Reportable, Watchable, Deletable, Nameable, Alertable, ChangeLoggable, Aliasable {
+  private static final Logger log = LoggerFactory.getLogger(Run.class);
+
   /** Field PREFIX */
   public static final String PREFIX = "RUN";
 
-  public void setId(long id);
+  private static final Pattern SOLID_RUN_REGEX = Pattern.compile("([A-z0-9\\-]+)_([0-9]{8})_(.*)");
+  public static final Long UNSAVED_ID = 0L;
+
+  public static Run createFromSolidXml(String statusXml, Function<String, SequencerReference> findSequencer) {
```

Comment:
This seems more confusing to create a class that doesn't follow the pattern. If you want a `RunUtils` class, that's fine.